ci/environment-3.8.yaml

```diff
@@ -17,7 +17,7 @@ dependencies:
   - pytest-cov
   - pytest-mock
   - python=3.8.*
-  - scikit-learn>=1.0.0
+  - scikit-learn <=1.1.3
```

Comment:
Why this version constraint?

Reply:
scikit-learn 1.2.0 has 123 test failures. I'm going follow up is that.

Reply:
Seems like a good reason! Thanks for clarifying